CHANGELOG.md

```diff
@@ -26,6 +26,53 @@ CHANGELOG
 Swift Next
 ----------
 
+* [SR-11298][]:
+
+  A class-constrained protocol extension, where the extended protocol does
+  not impose a class constraint, will now infer the constraint implicitly.
+
+  ```swift
+  protocol Foo {}
+  class Bar: Foo {
+    var someProperty: Int = 0
+  }
+
+  // Even though 'Foo' does not impose a class constraint, it is automatically
+  // inferred due to the Self: Bar constraint.
+  extension Foo where Self: Bar {
+    var anotherProperty: Int {
+      get { return someProperty }
+      // As a result, the setter is now implicitly nonmutating, just like it would
+      // be if 'Foo' had a class constraint.
+      set { someProperty = newValue }
+    }
+  }
+  ```
+  
+  As a result, this could lead to code that currently compiles today to throw an error.
+  
+  ```swift
+  protocol Foo {
+    var someProperty: Int { get set }
+  }
+  
+  class Bar: Foo {
+    var someProperty = 0
+  }
+  
+  extension Foo where Self: Bar {
+    var anotherProperty1: Int {
+      get { return someProperty }
+      // This will now error, because the protocol requirement
+      // is implicitly mutating and the setter is implicitly 
+      // nonmutating.
+      set { someProperty = newValue }
+    }
+  }
+  ```
+
+  To resolve this, explicitly mark the setter as `nonmutating` in the protocol.
```

Comment:
I still think this is the wrong recommendation: the extension is constrained because the protocol is not class-bound itself, and that means the mutating-ness of the existing property might be relevant.

Reply:
Sure, I suppose we can just drop this and just keep the non-working example. I thought it would make sense to offer at least one possible solution (perhaps this could be reworded), but it's probably okay to not offer one.

Reply:
No, I mean, the right answer is

```swift
var anotherProperty1: Int {
  get { return someProperty }
  set {
    var mutableSelf = self
    mutableSelf.someProperty = newValue
    assert(mutableSelf === self, "assigning the property really did reset 'self'")
  }
}
```

…though I'm not sure if we want to explain that to people.

Reply:
I think it's worth adding a solution.

Reply:
Added it now.